packages/esm-patient-chart-app/src/actions-buttons/add-past-visit.component.tsx

```diff
@@ -1,19 +1,16 @@
-import React from 'react';
+import { showModal } from '@openmrs/esm-framework';
+import React, { useCallback } from 'react';
 import { useTranslation } from 'react-i18next';
 
 interface AddPastVisitOverflowMenuItemProps {}
 
 const AddPastVisitOverflowMenuItem: React.FC<AddPastVisitOverflowMenuItemProps> = () => {
   const { t } = useTranslation();
-  const handleClick = React.useCallback(() => {
-    window.dispatchEvent(
-      new CustomEvent('visit-dialog', {
-        detail: {
-          type: 'prompt',
-          state: { type: 'past' },
-        },
-      }),
-    );
+
+  const openModal = useCallback(() => {
+    const dispose = showModal('start-visit-dialog', {
+      closeModal: () => dispose(),
+    });
```

Comment:
This is a funny pattern (using a function's return object in the callback for that same function), but I assume that its being used like this elsewhere and works as expected.